### DISCUSSION TITLE
Can HA support Hybrid - Local / Cloud Entities

### DISCUSSION THREAD

**Discussion by jeeftor** [home-assistant/architecture#977] [Entity Models]
Some integrations (that I've personally been involved with)

- IntelliFire
- WeatherFlow

Seem to have the ability to work both locally as well as in the cloud. For the WeatherFlow integration specifically you can pull local UDP weather data (which is whats in Home Assistant currently) - However - it also has a Cloud Based API which provides forecast as well as a slightly different view of sensor data. When you pull cloud data they will do a little aggregation of neary-by weather stations to add additional filters.

In the case of IntelliFire the APIs are nearly the same but there are a few features of the fireplace that are only available in the CloudAPI.

As it currently stands I think you can only be `Local/Cloud` and `Pollilng/Push`

What is the best way to approach a situation like this?

- Create two separate integrations:
   
   - `Weatherflow_local`
   - `Weatherflow_cloud`

- Single "Hybrid" Integration

  - Create an OptionsFlow to allow user to select where to push/pull data from

**Comment by thecode** [comment]
I have seen discussions about it in the past for different integrations and it was suggested to make a single integration, to make it simpler for users.
If an integration supports local push and cloud polling it should be marked as local push (the better from the available options).
For Shelly we didn't implement any options flow to select the mode of operation, we try local push and if it fails we switch to local polling (we didn't implement cloud although it is possible).
I think as long as it is a single integration the way you switch between modes or config it is up to you.

To sum up, we should only enforce creating a single integration, the rest is up to the specific case.

**Comment by Jc2k** [comment]
Quite often in this case you have 2 different sets of code owners and maybe even no shared code. A similar situation came up with airthings.

This lead to an architecture discussion and ultimately to integrations v2 and updates to the brands system.

https://github.com/home-assistant/core/pull/77284

https://github.com/home-assistant/architecture/discussions/794

https://github.com/home-assistant/core/pull/78801

Thanks to this effort we can have 2 different integrations with 2 different sets of code owners stepping on each other for overlapping sets of devices.

Personally imo if there's lots of shared code or a single set of maintainers, a single hybrid integration is fine. If it's different maintainers and the code doesn't really share anything (both of these apply to airthings) then 2 separate integrations linked to the same brand like airthings is fine.
